pkg/kubelet/pod_workers.go

```diff
@@ -63,22 +72,28 @@ type workUpdate struct {
 
 func newPodWorkers(dockerCache dockertools.DockerCache, syncPodFn syncPodFnType, recorder record.EventRecorder) *podWorkers {
 	return &podWorkers{
-		podUpdates:  map[types.UID]chan workUpdate{},
-		isWorking:   map[types.UID]bool{},
-		dockerCache: dockerCache,
-		syncPodFn:   syncPodFn,
-		recorder:    recorder,
+		podUpdates:       map[types.UID]chan workUpdate{},
+		isWorking:        map[types.UID]bool{},
+		desiredPodsCache: map[types.UID]*api.BoundPod{},
```

Comment:
We need to put workUpdates in the cache. The updateCompleteFn goes with the bound pod that we place (the use today is monitoring the total time from start -> running, we can't use the time from a previous invocation).